compiler+runtime/src/cpp/clojure/string_native.cpp

```diff
@@ -295,4 +295,69 @@ namespace clojure::string_native
 
     return make_box(s_str.substr(0, r));
   }
+
+  object_ref split(object_ref const s, object_ref const re)
+  {
+    auto const s_str(try_object<obj::persistent_string>(s)->data);
+    auto const regex(try_object<obj::re_pattern>(re)->regex);
+
+    std::string const search_str{ s_str.c_str() };
+
+    native_vector<object_ref> vec;
+    std::sregex_token_iterator iter(search_str.begin(), search_str.end(), regex, -1);
+    std::sregex_token_iterator end;
+
+    if(iter != end && iter->str().empty())
+    {
+      iter++;
```

Comment:
:wink:

Reply:
Ohh, I think I misunderstood your previous comment. I don't know why but I thought it wouldn't apply to iterators. Good to know that I should always prefer pre over post 👍

Reply:
It's specifically *because* of iterators, and other types which overload `++` and `--` that we need to be careful with them. They may not be cheap, since who knows how much state an iterator is holding? Sometimes it's a lot.